apps/system/lockscreen/js/source/timer_source.js

```diff
@@ -0,0 +1,68 @@
+/* global SourceEvent */
+'use strict';
+
+/**
+ * Event source for Stream. One Stream can collect events from multiple
+ * sources, which pass different native events (not only DOM events)
+ * to Stream.
+ **/
+(function(exports) {
+  var TimerSource = function(configs) {
```

Comment:
As a basic architecture I even want to check-in more sources to make a full abstraction layer. I'm afraid that people care dead code too much but the completed view too few, especially we couldn't land bunch of code with all use cases in one giant & complicated patch.